### PR TITLE
fix(client): Only show passwords while depressing the "show" button.

### PR DESCRIPTION
**Pull request by shane-tomlinson**
This builds on #3977 by only showing the password field if the show button is depressed.

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
cc @ryanfeeley 

Here's the behavior:

show-pw-on-press

**Comment by ryanfeeley** [comment]
@shane-tomlinson Looks great! If it alleviates issues on change password, should we change the behaviour there?

**Comment by shane-tomlinson** [comment]
@ryanfeeley - Thanks! I've added the behavior to both reset password and change password. I'm going to put this up on a dev box so you can play with it and we can try it on mobile. That might take until tomorrow, sometimes dev boxes take a bit of time to set up.

**Comment by shane-tomlinson** [comment]
This is running on https://stomlinson.dev.lcip.org. The behavior needs to be updated to correctly handle touch events on mobile devices.

**Comment by TDA** [comment]
@shane-tomlinson Is it supposed to be displayed only on click/depress or click/depress+hover? I am seeing some weirdly disconnected behavior:
stomlinpwdshow

I believe we would need a color scheme change too if we are changing the behavior to not trigger on :hover. The colors are confusing, and a few clicks seem to not be recognized.

**Comment by shane-tomlinson** [comment]
> Is it supposed to be displayed only on click/depress or click/depress+hover

Right now it's click/depress and stay within the button.

**Comment by shane-tomlinson** [comment]
@ryanfeeley - could you have a look at this behavior on https://stomlinson.dev.lcip.org? I have changed it where the user can depress the button and the show button is only hidden once the user lets up on the button, from anywhere in the page. I have also added touch events.

**Comment by ryanfeeley** [comment]
@shane-tomlinson It's hard to reproduce, but I can get it to show and stay shown with a single click.
click

**Comment by shane-tomlinson** [comment]
@philbooth - mind doing a review of this? 

The behavior seen by @ryanfeeley is strange. I am able to reproduce locally, but very carefully. I have to click _and not move the mouse afterwards_. Digging in, it seems the `mouseup` event is not always fired immediately when the element's type changes. Luckily, the problem is short lived, as soon as the mouse is moved, the mouseup event fires and the text is hidden again.

File: app/scripts/views/base.js [hunk 1]
```diff
@@ -679,33 +653,72 @@ define(function (require, exports, module) {
...
-        var self = this;
-        focusEl.one('focus', function () {
-          self.placeCursorAt(this, this.value.length);
-        });
-        focusEl.get(0).focus();
-      } catch (e) {
-        // IE can blow up if the element is not visible.
+      // Only focus if the device does not have touch. Setting focus
+      // on a touch device causes the virtual keyboard to pop up,
+      // which is jarring.
```
Comment (philbooth) [inline app/scripts/views/base.js]: As much as I love reading comments, we have the same comment here repeated just five lines after its original appearance. Is this one surplus to requirements?

Comment (shane-tomlinson) [inline app/scripts/views/base.js]: Silly me.

**Comment by philbooth** [comment]
LGTM, r+. 👍

**Pull request merged**